File: 11-container-with-most-water/11-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
             int area=0;
        int start=0;
        int end=height.size()-1;
      while(start!=end){
          
    int cur_area=min(height.at(start), height.at(end) )*(end-start);
          area=max(area,cur_area);
        //  if(area%(end-start)>max(height)) break;
        
          height[start] <= height[end] ? start ++ : end--; 
      }
           
           
       
        return area;
    }
};

File: 11-container-with-most-water/11-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l,r=0,len(height)-1
        Maxarea=0
        while(l<r):
            
            area=min(height[l],height[r])*(r-l)
            Maxarea=max(Maxarea,area)
            # print(height[l],height[r],l,r,area)
            
            if height[l] >= height[r] :
                r-=1
            elif height[l] <= height[r] :
                l+=1
            
            
        return Maxarea
                            